packages/ibm-products-web-components/src/components/coachmark/coachmark.ts

```diff
@@ -0,0 +1,182 @@
+
+  private handleClick = () => {
+    this._open = !this._open;
+  };
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this.dragCleanup) {
+      this.dragCleanup();
+    }
+    resetCoachmarkDetailsSignal();
+  }
+
+  private setupDraggable() {
+    const popover = this.shadowRoot?.querySelector(
+      '.my-popover'
+    ) as HTMLElement;
+    const popoverContent = popover.querySelector(
+      'cds-popover-content'
+    ) as HTMLElement;
+    const wrapper = popoverContent.querySelector(
+      '.draggable-wrapper'
+    ) as HTMLElement;
+    const slot = wrapper.querySelector('slot');
+    const assignedElements = slot?.assignedElements({ flatten: true });
+    const header = assignedElements?.find(
+      (el) => el.tagName.toLowerCase() === 'c4p-coachmark-header'
+    ) as HTMLElement;
+    requestAnimationFrame(() => {
+      const dragHandle = header.shadowRoot?.querySelector(
+        '.drag-handle'
+      ) as HTMLElement;
+
+      const draggable = makeDraggable({
+        el: popoverContent,
+        dragHandle: header,
+        focusableDragHandle: dragHandle,
+      });
+
+      const onDragStart = () => {
+        popoverContent.classList.add('is-dragging');
+        popoverContent.setAttribute(
+          'aria-label',
+          'Picked up the draggable popoverContent'
+        );
+      };
+      const onDragEnd = () => {
+        popoverContent.classList.remove('is-dragging');
+        popoverContent.setAttribute(
+          'aria-label',
+          'Draggable popoverContent was dropped'
+        );
+      };
+
+      popoverContent.addEventListener('dragstart', onDragStart);
+      popoverContent.addEventListener('dragend', onDragEnd);
+
+      // store cleanup
+      this.dragCleanup = () => {
+        popoverContent.removeEventListener('dragstart', onDragStart);
+        popoverContent.removeEventListener('dragend', onDragEnd);
+        draggable.cleanup?.();
+      };
+    });
+  }
+
+  firstUpdated() {
+    this.classList.add(blockClass);
+    if (this.floating) {
+      this.classList.add(`${blockClass}--floating`);
+      this.setupDraggable();
+      updateCoachmarkDetailsSignal({
+        name: 'floating',
+        detail: this.floating,
+      });
+    }
+  }
+
+  updated(changedProps: Map<string, unknown>) {
+    if (changedProps.has('position')) {
+      const { x = 0, y = 0 } = this.position ?? {};
+      if (x !== 0 || y !== 0) {
+        this.style.transform = `translate(${x}px, ${y}px)`;
+      }
+    }
+  }
+
+  render() {
+    return html`
+      <cds-popover
+        class="my-popover"
+        ?open=${this._open}
+        .caret=${this.floating === true ? false : true}
+        ?highContrast=${this.highContrast}
+        align=${this.align}
+        ?dropShadow=${this.dropShadow}
+        @click="${this.handleClick}"
+      >
+        <slot name="trigger"></slot>
+        <cds-popover-content>
+          <div class="draggable-wrapper">
```

Comment:
prefix. also, since `draggable` isn't the default behavior i think it makes more sense to give this a more generic name like `${prefix}-wrapper` or `${prefix}-container` and handle the draggable option with `classMap`